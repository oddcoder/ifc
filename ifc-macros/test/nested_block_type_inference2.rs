use ifc_macros::ifc_block;

fn main() {
    ifc_block!{
        #[IFC(Low)]
        let x = 5;
        #[IFC(Low)]
        let a = {
            let x = 4;
            x + 1
        };
    }
}