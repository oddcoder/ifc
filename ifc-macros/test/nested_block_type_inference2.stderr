error: Cannot assign high expression to low variable
  --> $DIR/nested_block_type_inference2.rs:8:9
   |
8  | /         let a = {
9  | |             let x = 4;
10 | |             x + 1
11 | |         };
   | |__________^
   |
help: Low Variable
  --> $DIR/nested_block_type_inference2.rs:8:13
   |
8  |         let a = {
   |             ^
help: High Expression
  --> $DIR/nested_block_type_inference2.rs:8:17
   |
8  |           let a = {
   |  _________________^
9  | |             let x = 4;
10 | |             x + 1
11 | |         };
   | |_________^
